apiserver/facades/agent/uniter/networkinfo.go

```diff
@@ -31,39 +30,34 @@ import (
 // Over time, the state types should be indirected, mocks generated, and
 // appropriate tests added.
 
-// PreferredAddressRetryArgs returns the retry strategy for getting a unit's
-// preferred address. Override for testing to use a different clock.
-// TODO (manadart 2019-10-09): Pass this as an argument to the NetworkInfo
-// constructor instead of exporting this public type for patching in tests.
-var PreferredAddressRetryArgs = func() retry.CallArgs {
-	return retry.CallArgs{
-		Clock:       clock.WallClock,
-		Delay:       3 * time.Second,
-		MaxDuration: 30 * time.Second,
-	}
-}
-
-// NetworkInfo is responsible for processing requests for network data for unit
-// endpoint bindings and/or relations.
+// NetworkInfo is responsible for processing requests for network data
+// for unit endpoint bindings and/or relations.
 type NetworkInfo struct {
-	st            *state.State
+	st *state.State
+	// retryFactory returns a retry strategy template used to poll for
+	// addresses that may not yet have landed in state,
+	// such as for CAAS containers or HA syncing.
+	retryFactory func() retry.CallArgs
```

Comment:
Just curious as I haven't seen this one before. Is this `retry` package a new thing? Is it meant to replace the retry logic (e.g. the exp backoff stuff) in other parts of the codebase?

Reply:
Doesn't appear to be new - it has been used here for quite some time. It doesn't appear to be a replacement for back-off strategies.